### DISCUSSION TITLE
Adding of new types of infill patterns

### DISCUSSION THREAD

**Discussion by andytikhon1994** [sn4k3/UVtools#880] [Q&A]
Hi!
Is it possible to add new types of infill patterns to UV tools?
For example, I want to add the array of straight lines which could be rotated by 90 degrees in neighboor layers (for simulating of true stereolithography printers pattern).

**Comment by sn4k3** [comment]
Hello,

You can send pictures of the pattern.
But already have gyroid which is one of the best infills and alternates rotation.

**Reply by andytikhon1994** [reply]
Do you have any hints about the exact application of infill spacing, thickness and wall thickness?


![Infill_pattern](https://github.com/sn4k3/UVtools/assets/171035915/3cdac11a-dd12-463c-907a-fecc2d08da4e)
The pattern looks like this picture. 
So, I want to control the thickness of white lines (1, 2 or more pixels) and the gap between white lines (black ones).

**Reply by sn4k3** [reply]
> Do you have any hints about the exact application of infill spacing, thickness and wall thickness?

Wall thickness is the thickness of model walls you want to have. Value is in pixels but you can convert mm to px. 3mm is standard in some slicers. Same goes for infill, is the thickness of the lines you want to have and spacing between them.
You need to play with values, and depends on model, but something like 3mm walls, 2mm infill is good.

> The pattern looks like this picture.

That infill pattern is weak and not good for resin printing, mostly you can't do it even/odd layer, because that will create lines of layer height which is too low/weak (0.05mm in most cases). The integrity of model is also bad with this pattern. 
Take a look at gyroid, it solve many problems of other infills, good XY integrity but also allow resin to drain (Not closed infill)

**Reply by andytikhon1994** [reply]
Thanks for your comment!

But I didn't understand the principle of implementation of infill thickness and infill spacing (especially, for gyroid structure). It is only clear for straight pillars pattern).
I usually work with DLP (LCD) and SLA printers, But with specific material - ceramic-filled resins. And usually we do not need to apply any infill tools during slicing (and all known for me slicers don't such kind of tool). But now I want to compare the behaviour of parts after DLP printing with the same parts after SLA printing. During SLA printing the polymerisation is made by the action of the laser with certain beam size and with the pattern of lines  (which show the movement of the laser) with certain space and pattern. These lines could be parallel, intersecting, concentric, etc. The simplest way - parallel lines in odd layer, and the same lines but rotated to 90 degrees in even layer.
Since I am not interested in only ceramic-filled resins but I want to burn out all organic content and sinter ceramics, infill pattern during printing could be critical for the integrity of the parts,

**Reply by sn4k3** [reply]
> But I didn't understand the principle of implementation of infill thickness and infill spacing (especially, for gyroid structure). It is only clear for straight pillars pattern).

No matter the pattern the infill thickness and spacing will have same behavior. It's just the thickness of each line/block and the spacing between each line. It will affect the straight pillars and gyroid the same way.
If still unclear the best thing to understand is visualize (brain learn best by experimentation and visualize by itself), open same file in other instance and run two infill with different settings, eg: infill thickness 2 vs 20. This way you will understand the impact of each setting. 

> and all known for me slicers don't such kind of tool

There are reasons for that, it can cause more trouble than good... Because most of infills will create resin traps, not easy to clean all resin from inside, but also "impossible" to cure all inside structure, and uncured resin is not good to have. So the recommendation is either solid or hollow. Infill for resin must be used with care, but definitely gyroid is one of the best for integrity. You can read more [here](https://bigrep.com/posts/gyroid-infill-3d-printing)
